python/sidecar/projects.py

```diff
@@ -106,7 +107,62 @@ def delete_project(user_id: str, project_id: str) -> None:
         pass
 
 
-def get_project_files(user_id: str, project_id: str) -> list:
+def get_project_files(user_id: str, project_id: str) -> ProjectFileTreeResponse:
+    """
+    Get the files in a project as a tree structure.
+
+    Returns
+    -------
+    ProjectFileTreeResponse
+        A tree structure of the files in a project.
+    """
     project_path = get_project_path(user_id, project_id)
-    filepaths = list(project_path.glob("**/*"))
-    return filepaths
+    project_name = get_project_name(user_id, project_id)
+
+    if not project_path.exists():
+        raise FileNotFoundError(f"Project {project_id} does not exist")
+
+    contents = []
+    for obj in project_path.rglob("*"):
+        relpath = obj.relative_to(project_path)
+
+        # Skip hidden files and directories
+        if relpath.name.startswith(".") or obj.name.startswith("."):
+            continue
+
+        # files in subdirectories
+        if obj.is_dir():
+            contents.append(create_folder_entry(obj, project_path=project_path))
+            continue
```

Comment:
Had to google what that trap was. Wouldn't this be the opposite of that though since `continue` jumps to the next item in the iteration?

I can change though.